src/components/structures/RoomStatusBar.tsx

```diff
@@ -93,19 +94,27 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
             syncStateData: this.context.getSyncStateData(),
             unsentMessages: getUnsentMessages(this.props.room),
             isResending: false,
+            timelineNeedsRefresh: this.props.room.getTimelineNeedsRefresh(),
         };
     }
 
     public componentDidMount(): void {
         const client = this.context;
         client.on("sync", this.onSyncStateChange);
         client.on("Room.localEchoUpdated", this.onRoomLocalEchoUpdated);
+        this.props.room.on(RoomEvent.historyImportedWithinTimeline, this.onRoomHistoryImportedWithinTimeline);
```

Comment:
> Please add the context type definition around line L87 as string emitter keys are no longer allowed in TypedEventEmitter
>
> *-- https://github.com/matrix-org/matrix-react-sdk/pull/8303#discussion_r850187564&

I'm not sure what to do here exactly. I've stopped using string keys. Is there anything further?

Reply:
It has fallen out of scope as you're no longer touching `this.context`